packages/react-components/lib/tasks/task-timeline.tsx

```diff
@@ -48,47 +47,99 @@ const StyledTimeLine = styled((props: TimeLinePropsWithRef) => <Timeline {...pro
   }),
 );
 
+function nestedEvents(eventsStates: { [key: string]: EventState } | undefined, child: number) {
+  if (eventsStates !== undefined) {
+    const deps = eventsStates[child] ? eventsStates[child].deps : [];
+    return (
+      <TreeItem
+        key={`event-${child}`}
+        nodeId={`parent-${eventsStates[child].name}`}
+        label={eventsStates[child].name}
+      >
+        {deps?.map((id) => {
+          if (eventsStates[id].deps?.length) {
+            nestedEvents(eventsStates, id);
+          } else {
+            return (
+              <TreeItem
+                key={eventsStates[child].name}
+                nodeId={`child-${id}`}
+                label={eventsStates[id].name}
+              ></TreeItem>
+            );
+          }
+        })}
+      </TreeItem>
+    );
+  }
+}
+
+function colorDot(eventsStates: { [key: string]: EventState } | undefined) {
+  let standby: Boolean = true;
+  for (const id in eventsStates) {
+    const event: EventState = eventsStates[id];
+    if (event.status === 'underway') {
+      return 'success' as const;
+    }
+    if (event.status !== 'standby') {
+      standby = false;
+    }
+  }
+  if (standby) return 'info' as const;
+
+  return 'error' as const;
+}
+
 export interface TaskTimelineProps {
   taskState: TaskState;
 }
 
 export function TaskTimeline({ taskState }: TaskTimelineProps): JSX.Element {
   const phases = taskState.phases ? Object.values(taskState.phases) : [];
-
   return (
     <StyledTimeLine className={classes.timelineRoot}>
-      {phases.map((phase, idx) => (
-        <TimelineItem key={phase.id}>
-          <TimelineOppositeContent style={{ flex: 1, padding: '2px 2px' }}>
-            <Typography variant="overline" color="textSecondary" style={{ textAlign: 'justify' }}>
-              {phase.unix_millis_start_time
-                ? new Date(phase.unix_millis_start_time).toLocaleTimeString()
-                : 'unknown'}
-            </Typography>
-          </TimelineOppositeContent>
-          <TimelineSeparator>
-            <TimelineDot color="info" />
-            {idx < phases.length - 1 && <TimelineConnector />}
-          </TimelineSeparator>
-          <TimelineContent>
-            <TreeView
-              defaultCollapseIcon={<ExpandMoreIcon />}
-              defaultExpandIcon={<ChevronRightIcon />}
-            >
-              {/* FIXME: rmf does not return event hierarchy information */}
-              {phase.events
-                ? Object.values(phase.events).map((event) => (
-                    <TreeItem
-                      key={event.id}
-                      nodeId={event.id.toString()}
-                      label={event.name || 'unknown'}
-                    ></TreeItem>
-                  ))
-                : null}
-            </TreeView>
-          </TimelineContent>
-        </TimelineItem>
-      ))}
+      {phases.map((phase, idx) =>
+        phase.events ? (
```

Comment:
Are you sure we should hide phases without events?

Reply:
Nope I'm note sure, it was not specified in the figma if we should or should not.
Do you think we should show them?

Reply:
I'm not sure what kind of phases will have no events, and if this will be a good heuristics to hide possibly uninteresting data from the user, maybe @mxgrey can give better insights.